### PR TITLE
[NOD-816] Remove TxIndex and AddrIndex

### PR DESCRIPTION
**Pull request by stasatdaglabs**

### CODE REVIEW

File: cmd/addsubnetwork/addsubnetwork.go [hunk 1]
```diff
@@ -1,88 +0,0 @@
-package main
```
Comment (svarogg) [inline cmd/addsubnetwork/addsubnetwork.go]: Why did you delete addsubnetwork?
Doesn't seem related to all the indexers...

File: server/rpc/handle_get_tx_out.go [hunk 2]
```diff
@@ -109,10 +99,9 @@ func handleGetTxOut(s *Server, cmd interface{}, closeChan <-chan struct{}) (inte
 	}
 
 	txOutReply := &rpcmodel.GetTxOutResult{
-		SelectedTip:   selectedTipHash,
-		Confirmations: confirmations,
```
Comment (svarogg) [inline server/rpc/handle_get_tx_out.go]: Since you are taking the data from the utxo-set, can't you calculate confirmations?
(Same things for other places where you deleted confirmations out of rpc results)/

File: server/rpc/rpcserverhelp.go [hunk 3]
```diff
@@ -223,28 +223,13 @@ var helpDescsEnUS = map[string]string{
 	"txRawResult-vin":           "The transaction inputs as JSON objects",
 	"txRawResult-vout":          "The transaction outputs as JSON objects",
 	"txRawResult-blockHash":     "Hash of the block the transaction is part of",
-	"txRawResult-confirmations": "Number of confirmations of the block (Will be 'null' if txindex is not disabled)",
+	"txRawResult-confirmations": "Number of confirmations of the block",
```
Comment (svarogg) [inline server/rpc/rpcserverhelp.go]: It looks like you deleted this everywhere, shouldn't it be delete there as well?
I did ask you to maybe restore it at least in part of the places, so this comment is only relevant in case we decide that restoring is not an option.

File: server/rpc/rpcserverhelp.go [hunk 4]
```diff
@@ -478,7 +455,7 @@ var helpDescsEnUS = map[string]string{
 
 	// GetTxOutResult help.
 	"getTxOutResult-selectedTip":   "The block hash that contains the transaction output",
-	"getTxOutResult-confirmations": "The number of confirmations (Will be 'null' if txindex is not disabled)",
+	"getTxOutResult-confirmations": "The number of confirmations",
```
Comment (svarogg) [inline server/rpc/rpcserverhelp.go]: This one too

File: blockdag/dag.go [hunk 5]
```diff
@@ -1366,6 +1366,23 @@ func (dag *BlockDAG) BlockConfirmationsByHashNoLock(hash *daghash.Hash) (uint64,
 	return dag.blockConfirmations(node)
 }
 
+// TxConfirmations returns the confirmations for the given outpoint, if it exists
+// in the DAG.
```
Comment (svarogg) [inline blockdag/dag.go]: "if it exists in the DAG" - not exactly. It might be already spend in the DAG.
I'd say something like "if it exists in the DAG's UTXO-Set" or something.

File: blockdag/dag.go [hunk 6]
```diff
@@ -1366,6 +1366,23 @@ func (dag *BlockDAG) BlockConfirmationsByHashNoLock(hash *daghash.Hash) (uint64,
...
+//
+// This function is safe for concurrent access.
+func (dag *BlockDAG) TxConfirmations(outpoint *wire.Outpoint) (uint64, bool) {
```
Comment (svarogg) [inline blockdag/dag.go]: Rename to UTXO confirmations or something, since it's never dealing with whould Txs.

File: cmd/addblock/config.go [hunk 7]
```diff
@@ -40,12 +40,10 @@ func ActiveConfig() *ConfigFlags {
...
 // See loadConfig for details on the configuration load process.
 type ConfigFlags struct {
-	DataDir   string `short:"b" long:"datadir" description:"Location of the kaspad data directory"`
-	DbType    string `long:"dbtype" description:"Database backend to use for the Block DAG"`
-	InFile    string `short:"i" long:"infile" description:"File containing the block(s)"`
-	TxIndex   bool   `long:"txindex" description:"Build a full hash-based transaction index which makes all transactions available via the getrawtransaction RPC"`
-	AddrIndex bool   `long:"addrindex" description:"Build a full address-based transaction index which makes the searchrawtransactions RPC available"`
-	Progress  int    `short:"p" long:"progress" description:"Show a progress message each time this number of seconds have passed -- Use 0 to disable progress announcements"`
+	DataDir  string `short:"b" long:"datadir" description:"Location of the kaspad data directory"`
+	DbType   string `long:"dbtype" description:"Database backend to use for the Block DAG"`
```
Comment (someone235) [inline cmd/addblock/config.go]: Fix to DBType

File: cmd/addblock/config.go [hunk 8]
```diff
@@ -40,12 +40,10 @@ func ActiveConfig() *ConfigFlags {
...
+	InFile   string `short:"i" long:"infile" description:"File containing the block(s)"`
+	Progress int    `short:"p" long:"progress" description:"Show a progress message each time this number of seconds have passed -- Use 0 to disable progress announcements"`
```
Comment (someone235) [inline cmd/addblock/config.go]: Add AcceptanceIndex flag

File: server/rpc/rpcserverhelp.go (hunk 3 above)
Comment (someone235) [inline server/rpc/rpcserverhelp.go]: This is not used. Please remove this field

File: cmd/addblock/config.go [hunk 9]
```diff
@@ -40,10 +40,11 @@ func ActiveConfig() *ConfigFlags {
...
 type ConfigFlags struct {
-	DataDir  string `short:"b" long:"datadir" description:"Location of the kaspad data directory"`
-	DbType   string `long:"dbtype" description:"Database backend to use for the Block DAG"`
-	InFile   string `short:"i" long:"infile" description:"File containing the block(s)"`
-	Progress int    `short:"p" long:"progress" description:"Show a progress message each time this number of seconds have passed -- Use 0 to disable progress announcements"`
+	DataDir         string `short:"b" long:"datadir" description:"Location of the kaspad data directory"`
+	DBType          string `long:"dbtype" description:"Database backend to use for the Block DAG"`
+	InFile          string `short:"i" long:"infile" description:"File containing the block(s)"`
+	Progress        int    `short:"p" long:"progress" description:"Show a progress message each time this number of seconds have passed -- Use 0 to disable progress announcements"`
+	AcceptanceIndex bool   `long:"acceptanceindex" description:"Maintain a full hash-based acceptance index which makes the getChainByBlock RPC available"`
```
Comment (someone235) [inline cmd/addblock/config.go]: getChainByBlock -> getChainFromBlock
